src/key_reset.c

```diff
@@ -901,15 +904,22 @@ PEP_STATUS create_standalone_key_reset_message(PEP_SESSION session,
                 && recip && ! EMPTYSTR(recip->address)
                 && ! EMPTYSTR(old_fpr) && ! EMPTYSTR(new_fpr));
 
+
+    PEP_STATUS status;
+
     *dst = NULL;
     
     message* reset_msg = NULL;
-    
-    PEP_STATUS status = _generate_keyreset_command_message(session, own_identity,
-                                                           recip,
-                                                           old_fpr, new_fpr, false,
-                                                           &reset_msg);
-                            
+
+    own_identity = new_identity(own_identity->address, old_fpr, PEP_OWN_USERID, own_identity->username);
+
+    status = set_own_key(session, own_identity, old_fpr);
```

Comment:
This writes to the DB and triggers certain sync events. Can we avoid it?

Reply:
Yup. Is avoidable, fixed in PR2 which is following soon